server/src/main/java/org/opensearch/search/DefaultSearchContext.java

```diff
@@ -879,8 +881,20 @@ public Profilers getProfilers() {
      * Returns concurrent segment search status for the search context
      */
     @Override
-    public boolean isConcurrentSegmentSearchEnabled() {
-        return useConcurrentSearch;
+    public boolean shouldUseConcurrentSearch() {
+        // concurrentSearchSettingsEnabled is evaluated when searchContext is created
+        return concurrentSearchSettingsEnabled && requestShouldUseConcurrentSearch;
+    }
+
+    public void evaluateRequestShouldUseConcurrentSearch() {
+        requestShouldUseConcurrentSearch = true;
+        if (aggregations() != null && aggregations().factories() != null) {
+            for (AggregatorFactory factory : aggregations().factories().getFactories()) {
+                if (factory.traverseChildFactories() == false) {
+                    requestShouldUseConcurrentSearch = false;
```

Comment:
With the recommendation provided below we can change this to just do `return aggregations().factories(). allFactoriesSupportConcurrentSearch()`